audiorecorder/src/main/java/org/odk/collect/audiorecorder/recording/AudioRecorderService.kt

```diff
@@ -41,10 +41,23 @@ class AudioRecorderService : Service() {
         when (intent?.action) {
             ACTION_START -> {
                 val sessionId = intent.getSerializableExtra(EXTRA_SESSION_ID)
-                val output = intent.getSerializableExtra(EXTRA_OUTPUT) as Output
+                lastUsedOutput = intent.getSerializableExtra(EXTRA_OUTPUT) as Output
 
                 if (!recorder.isRecording() && sessionId != null) {
-                    startRecording(sessionId, output)
+                    startRecording(sessionId, lastUsedOutput)
+                }
+            }
+
+            ACTION_RESTART -> {
+                val file = recorder.stop()
+                recordingRepository.recordingReady(file)
```

Comment:
One problem I can see here is that this will end up calling `RecordingHandler#handle` ([here](https://github.com/getodk/collect/blob/4ec8ed26aafde5e621268999b36ee17cbfaed882/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java#L529)), which will then call `FormSaveViewMode#resumeSave`. This is because before when saving, the order of events would always be:

1. User saves through UI
2. Recording is happening, so recording is saved as media file
3. Saving is resumed
4. Form exits

With this new structure, the form save goes ahead (in `FormSaveViewModel#saveForm` simultaneously. There are a couple of risks with that:

1. The form could save before the recording has been saved as an answer file or before the question is answered with that file meaning the instance would incorrect.
2. The form could end up being saved twice which risks two save tasks running concurrently

I think we can alleviate these risks by also using the `WAITING_TO_SAVE` state when we do a `restart` in `FormSaveViewModel#saveForm`, basically following the same structure as when we are exiting after save.

How saving the form and saving the recording interact is pretty convoluted at the moment (a lot of the complexity comes from needing to set an answer on the `FormEntryController` before saving to disk I guess). I'd be interested to see if you have any ideas for making this nicer!